src/dotnet/PrintableTable.cs

```diff
@@ -0,0 +1,199 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using Microsoft.DotNet.Tools;
+
+namespace Microsoft.DotNet.Cli
+{
+    // Represents a table (with rows of type T) that can be printed to a terminal.
+    internal class PrintableTable<T>
+    {
+        private const string ColumnDelimiter = "      ";
+        private List<Column> _columns = new List<Column>();
+
+        private class Column
+        {
+            public string Header { get; set; }
+            public Func<T, string> GetContent { get; set; }
+            public int MaxWidth { get; set; }
+            public override string ToString() { return Header; }
+        }
+
+        public void AddColumn(string header, Func<T, string> getContent, int maxWidth = int.MaxValue)
+        {
+            if (string.IsNullOrEmpty(header))
```

Comment:
Why disallow an empty header?

Reply:
No reason, really.  Will need to change at least one place that expects the calculated column width to never be 0 if we want it.

Reply:
I'll be pushing an update that supports empty headers and handles zero-width columns.